helm/charts/hpe-csi-driver/templates/_helpers.tpl

```diff
@@ -30,3 +30,38 @@ Create chart name and version as used by the chart label.
 {{- define "hpe-csi-storage.chart" -}}
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+CHAP secret validation
+*/}}
+{{- define "hpe-csi-storage.chapSecretValidation" -}}
+{{- if not (empty .Values.iscsi.chapSecretName) }}
+  {{- $secret := lookup "v1" "Secret" .Release.Namespace .Values.iscsi.chapSecretName }}
+  {{- if not $secret }}
+    {{- fail (printf "Secret %s not found in namespace %s" .Values.iscsi.chapSecretName .Release.Namespace) }}
+  {{- end }}
+
+  {{- $username := index $secret.data "chapUser" | b64dec }}
+  {{- $password := index $secret.data "chapPassword" | b64dec }}
+
+  {{- if or (empty $username) (empty $password) }}
+    {{- fail "Username or password cannot be empty." }}
+  {{- end }}
+
+  {{- $chapUserValidationPattern := "^[a-zA-Z0-9][a-zA-Z0-9\\-:.]{0,63}$" }}
+  {{- $chapPasswordValidationPattern := "^$|^[a-zA-Z0-9+_)(*^%$#@!]{12,16}$" }}
```

Comment:
Isn't this regex a bit too strict? I looked at the Nimble REST API docs: 

```
CHAP secret.The CHAP secret should be between 12-16 characters and cannot contain spaces or most punctuation. String of 12 to 16 printable ASCII characters excluding ampersand and ^[];`. Example: 'password_25-24
```

Reply:
This is the same pattern we used earlier for validating the secret - https://github.com/hpe-storage/co-deployments/blob/cfe36d87211e34d6ca9f77690d24fb98a887922a/helm/charts/hpe-csi-driver/values.schema.json

Reply:
Old code is old sins. I'm just saying that the CHAP password we use as an example in the platform docs should work with the CSI driver.

Reply:
Agreed, there were other issues I found when I reviewed this in detail. Thank you for checking. I've updated the regex to match the description and made it more readable.